File: backend/app/models/feedback.py
from pydantic import BaseModel


class FeedbackCreate(BaseModel):
    solution_id: int
    rating: int
    feedback_type: str = "general"
    comments: str = ""


class FeedbackResponse(BaseModel):
    id: str
    user_id: str
    solution_id: str
    rating: int
    feedback_type: str
    comments: str
    created_at: str

File: frontend/src/types/feedback.ts
export interface Feedback {
  id: string;
  solutionId: string;
  rating: number;
  feedbackType: string;
  comments: string;
  createdAt: string;
}

export interface FeedbackCreate {
  solution_id: number;
  rating: number;
  feedback_type?: string;
  comments?: string;
}
